recipe/patches/0009-add-GRPC_DLL-mechanism-for-missing-symbols-on-window.patch

```diff
@@ -0,0 +1,154 @@
+   include/grpc/byte_buffer_reader.h
+diff --git a/include/grpc/impl/codegen/port_platform.h b/include/grpc/impl/codegen/port_platform.h
+index ae0e35cfb7..63baf21c6a 100644
+--- a/include/grpc/impl/codegen/port_platform.h
++++ b/include/grpc/impl/codegen/port_platform.h
+@@ -89,6 +89,25 @@
+ #undef GRPC_NOMINMAX_WAS_NOT_DEFINED
+ #undef NOMINMAX
+ #endif /* GRPC_WIN32_LEAN_AND_MEAN_WAS_NOT_DEFINED */
++
++// GPRC_DLL
++// inspired by https://github.com/abseil/abseil-cpp/blob/20220623.1/absl/base/config.h#L730-L747
++//
++// When building gRPC as a DLL, this macro expands to `__declspec(dllexport)`
++// so we can annotate symbols appropriately as being exported. When used in
++// headers consuming a DLL, this macro expands to `__declspec(dllimport)` so
++// that consumers know the symbol is defined inside the DLL. In all other cases,
++// the macro expands to nothing.
++// Note: gRPC_DLL_EXPORTS is set in CMakeLists.txt when building shared grpc{,_unsecure}
++//       gRPC_DLL_IMPORTS is set by us as part of the interface for consumers of the DLL
++#if defined(gRPC_DLL_EXPORTS)
++#define GRPC_DLL __declspec(dllexport)
++#elif defined(gRPC_DLL_IMPORTS)
++#define GRPC_DLL __declspec(dllimport)
++#else
++#define GRPC_DLL
++#endif // defined(gRPC_DLL_EXPORTS)
++
+ #endif /* defined(_WIN64) || defined(WIN64) || defined(_WIN32) || \
+           defined(WIN32) */
+ 
+diff --git a/src/core/lib/gprpp/fork.h b/src/core/lib/gprpp/fork.h
+index d6ef16130a..46b067c140 100644
+--- a/src/core/lib/gprpp/fork.h
++++ b/src/core/lib/gprpp/fork.h
+@@ -35,7 +35,7 @@ class ExecCtxState;
+ class ThreadState;
+ }  // namespace internal
+ 
+-class Fork {
++class GRPC_DLL Fork {
+  public:
+   typedef void (*child_postfork_func)(void);
+ 
+diff --git a/third_party/upb/upb/port_def.inc b/third_party/upb/upb/port_def.inc
+index 2b240ff6e8..dd71d745e9 100644
+--- a/third_party/upb/upb/port_def.inc
++++ b/third_party/upb/upb/port_def.inc
+@@ -259,3 +259,12 @@ void __asan_unpoison_memory_region(void const volatile *addr, size_t size);
+ #else
+ #define UPB_TREAT_PROTO2_ENUMS_LIKE_PROTO3 0
+ #endif
++
++// More details in include/grpc/impl/codegen/port_platform.h
++#if defined(gRPC_DLL_EXPORTS)
++#define GRPC_DLL __declspec(dllexport)
++#elif defined(gRPC_DLL_IMPORTS)
++#define GRPC_DLL __declspec(dllimport)
++#else
++#define GRPC_DLL
++#endif // defined(gRPC_DLL_EXPORTS)
+diff --git a/third_party/upb/upb/upb.c b/third_party/upb/upb/upb.c
+index 55730bbe82..18c0067da6 100644
+--- a/third_party/upb/upb/upb.c
++++ b/third_party/upb/upb/upb.c
+@@ -112,7 +112,11 @@ static uintptr_t upb_cleanup_metadata(uint32_t* cleanup,
+   return (uintptr_t)cleanup | has_initial_block;
+ }
+ 
+-upb_alloc upb_alloc_global = {&upb_global_allocfunc};
++// For some reason, this file does not include upb.h where the function
++// declaration with the dllexport/import spec is. Need to redefine
++// with the dllexport spec
++
++GRPC_DLL upb_alloc upb_alloc_global = {&upb_global_allocfunc};
```

Comment:
Where is `GRPC_DLL` defined for this file?

Reply:
Through `port_def.inc` (see above), which is included in both [`upb.c`](https://github.com/grpc/grpc/blob/v1.48.1/third_party/upb/upb/upb.c#L40) & [`upb.h`](https://github.com/grpc/grpc/blob/v1.48.1/third_party/upb/upb/upb.h#L42)